### PR TITLE
[action] carthage - add use_xcframeworks option

### PR DESCRIPTION
**Pull request by att55**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description

Carthage has --use-xcframeworks option since ver.0.37.0.
So, I also added the option to CarthageAction.

https://github.com/Carthage/Carthage/releases/tag/0.37.0

### CODE REVIEW

File: fastlane/spec/actions_specs/carthage_spec.rb [hunk 1]
```diff
@@ -666,6 +686,53 @@
...
+        context "when command is bootstrap" do
+          let(:command) { 'bootstrap' }
+
+          it "adds the use_xcframeworks option" do
+            result = Fastlane::FastFile.new.parse("lane :test do
+                carthage(command: '#{command}', use_xcframeworks: true)
+              end").runner.execute(:test)
+            expect(result).to eq("carthage bootstrap --use-xcframeworks")
+          end
+        end
```
Comment (ainame) [inline fastlane/spec/actions_specs/carthage_spec.rb:734]:
```suggestion
        context "when valid command is used for --use-xcframeworks option" do
          it "adds the use_xcframeworks option" do
            ['update', 'build', 'bootstrap'].each do |command|
              result = Fastlane::FastFile.new.parse("lane :test do
                carthage(command: '#{command}', use_xcframeworks: true)
              end").runner.execute(:test)
              expect(result).to eq("carthage bootstrap --use-xcframeworks")
            end
          end
        end
```

I think you can combine those cases into one case. Can you please do that?

**Review by ainame — CHANGES_REQUESTED** [review]
This is same as https://github.com/fastlane/fastlane/pull/18077 
Can you please add `type` and shrink test cases?

File: fastlane/spec/actions_specs/carthage_spec.rb (hunk 1 above)
Comment (att55) [reply fastlane/spec/actions_specs/carthage_spec.rb:734]: @ainame 
Thanks for your review and suggestion! 
But, I think the expectation should be like below.  
```
expect(result).to eq("carthage bootstrap --use-xcframeworks")
```
Except for that, I will take your suggestion in PR 👍

**Comment by att55** [comment]
@ainame Added type and combined test cases in one case 👍 
Please check when you're available.

File: fastlane/spec/actions_specs/carthage_spec.rb [hunk 2]
```diff
@@ -665,6 +685,17 @@
             end
           end
         end
+
+        context "when valid command is used for --use-xcframeworks option" do
```
Comment (ainame) [inline fastlane/spec/actions_specs/carthage_spec.rb:689]:
```suggestion
        context "when --use-xcframeworks option is present  with valid command" do
```

**Review by ainame — APPROVED** [review]
LGTM👍  @joshdholtz will double-check and merge this when he can 🙇

**Comment by att55** [comment]
@ainame It seems like one job is failed.
https://app.circleci.com/pipelines/github/fastlane/fastlane/2080/workflows/612d07b0-3f6b-4ee9-9d6d-9a6e00e59c7b/jobs/61174

**Comment by ainame** [comment]
It's a known flaky test case. Don't worry.

**Review by joshdholtz — APPROVED** [review]
Thanks again for another carthage addition! Really appreciate it ❤️

**Pull request merged**